spec/acceptance/disable_new_tasking_system_spec.rb

```diff
@@ -0,0 +1,141 @@
+require 'spec_helper_acceptance'
+
+describe 'initial configuration with new tasking system' do
+  certdir = '/etc/pulpcore-certs'
+
+  it_behaves_like 'an idempotent resource' do
+    let(:manifest) do
+      <<-PUPPET
+      class { 'pulpcore':
+        worker_count              => 1,
+        use_legacy_tasking_system => false,
+      }
+      PUPPET
+    end
+  end
+
+  describe service('httpd') do
+    it { is_expected.to be_enabled }
+    it { is_expected.to be_running }
+  end
+
+  describe service('pulpcore-api') do
+    it { is_expected.to be_enabled }
+    it { is_expected.to be_running }
+  end
+
+  describe service('pulpcore-content') do
+    it { is_expected.to be_enabled }
+    it { is_expected.to be_running }
+  end
+
+  describe service('pulpcore-resource-manager') do
+    it { is_expected.to be_enabled }
+    it { is_expected.to be_running }
+  end
+
+  describe service('pulpcore-worker@1') do
+    it { is_expected.to be_enabled }
+    it { is_expected.to be_running }
+  end
+
+  describe port(80) do
+    it { is_expected.to be_listening }
+  end
+
+  describe port(443) do
+    it { is_expected.to be_listening }
+  end
+
+  describe curl_command("https://#{host_inventory['fqdn']}/pulp/api/v3/status/", cacert: "#{certdir}/ca-cert.pem") do
+    its(:response_code) { is_expected.to eq(200) }
+    its(:exit_status) { is_expected.to eq 0 }
+  end
+
+  describe command("PULP_SETTINGS=/etc/pulp/settings.py pulpcore-manager diffsettings") do
+    its(:stdout) { is_expected.to match(/^USE_NEW_WORKER_TYPE = True/) }
+    its(:exit_status) { is_expected.to eq 0 }
+  end
+
+  describe command("DJANGO_SETTINGS_MODULE=pulpcore.app.settings PULP_SETTINGS=/etc/pulp/settings.py rq info -c pulpcore.rqconfig") do
+    its(:stdout) { is_expected.to match(/^0 workers, /) }
+    its(:stdout) { is_expected.not_to match(/^resource-manager /) }
+    its(:exit_status) { is_expected.to eq 0 }
+  end
+
+  describe command("ps -e -o command") do
```

Comment:
Would [process](https://serverspec.org/resource_types.html#process) be a better type to use?

Reply:
The part I struggle with is this bit: "When several processes match, only the parameters of the first one are available" so this could return the parameters of any worker or even the resource manager -- as the new pulpcore-worker and resource manager share the same executable name and the resource manager is differentiated by appending the `--resource-manager` argument.

That may not be a big deal, as the new "resource manager" as I understand it is a dummy process which does nothing, is not required for the new tasking system to function, and seems to exist to allow switching between the two tasking systems without breaking tools like foreman-maintain that expect to find a `pulpcore-resource-manager.service`

The way I have it currently written tests that at least one resource manager and at least one non resource manager worker are running. I think I wouldn't be able to do that with https://serverspec.org/resource_types.html#process

Reply:
Looking back at this with a better understanding of https://pulp.plan.io/issues/8721 , this test might be entirely redundant anyway.